src/main/java/com/banking/bankservice/service/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.banking.bankservice.service.impl;
+
+import com.banking.bankservice.model.Account;
+import com.banking.bankservice.model.Transaction;
+import com.banking.bankservice.repository.TransactionRepository;
+import com.banking.bankservice.service.AccountService;
+import com.banking.bankservice.service.TransactionService;
+import java.time.LocalDateTime;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TransactionServiceImpl implements TransactionService {
+    private final TransactionRepository transactionRepository;
+    private final AccountService accountService;
+
+    @Autowired
+    public TransactionServiceImpl(TransactionRepository transactionRepository,
+                                  AccountService accountService) {
+        this.transactionRepository = transactionRepository;
+        this.accountService = accountService;
+    }
+
+    @Override
+    public List<Transaction> getAllByAccount(int page, int size, Account account) {
+        Pageable pageable = PageRequest.of(page, size, Sort.by("date").descending());
+        return transactionRepository.getAllByAccountNumber(account, pageable);
+    }
+
+    @Override
+    public void transfer(Account accountFrom, Account accountTo, Double amount) {
```

Comment:
You can use `@Transactional` annotation over this method

Reply:
done